File: backend/models/noble_phantasm.py
from pydantic import BaseModel


class NoblePhantasmEffect(BaseModel):
    tier: str | None = None
    tier_level: int | None = None
    description: str


class NoblePhantasmSkill(BaseModel):
    level: int
    tier: str | None = None
    tier_level: int | None = None
    description: str


class NoblePhantasm(BaseModel):
    name: str
    character: str | None = None
    is_global: bool
    lore: str
    effects: list[NoblePhantasmEffect]
    skills: list[NoblePhantasmSkill]

File: frontend/src/types/noble-phantasm.ts
export interface NoblePhantasmEffect {
  tier: string | null;
  tier_level: number | null;
  description: string;
}

export interface NoblePhantasmSkill {
  level: number;
  tier: string | null;
  tier_level: number | null;
  description: string;
}

export interface NoblePhantasm {
  name: string;
  character: string | null;
  is_global: boolean;
  lore: string;
  effects: NoblePhantasmEffect[];
  skills: NoblePhantasmSkill[];
  last_updated?: number;
}
